GitCommands/Git/GitModule.cs

```diff
@@ -950,11 +950,15 @@ public GitRevision GetRevision(ObjectId? objectId = null, bool shortFormat = fal
             }
             else
             {
-                string message = ProcessDiffNotes(10);
+                string message = ProcessDiffNotes(startIndex: 10);
 
                 // commit message is not re-encoded by git when format is given
-                revision.Body = ReEncodeCommitMessage(message, revision.MessageEncoding);
-                revision.Subject = revision.Body?.Substring(0, revision.Body.IndexOfAny(new[] { '\r', '\n' })) ?? "";
+                // See also RevisionReader for parsing commit body
+                revision.Body = ReEncodeCommitMessage(message, revision.MessageEncoding)?.TrimStart().TrimEnd();
```

Comment:
Why not 
```suggestion
                revision.Body = ReEncodeCommitMessage(message, revision.MessageEncoding)?.Trim();
```

Also why not trim inside `ReEncodeCommitMessage`?

Reply:
That affects a number of other calls too. I believe this is OK, changing.
It though means that GE _never_ present commit body as entered by the user.
But it is better that handling is consistent, this is what this part of the PR is all about.